packages/sanity/src/desk/panes/document/DocumentPaneProvider.tsx

```diff
@@ -119,22 +114,38 @@ export const DocumentPaneProvider = memo((props: DocumentPaneProviderProps) => {
     params.path ? pathFromString(params.path) : []
   )
   const activeViewId = params.view || (views[0] && views[0].id) || null
-  const timeline = useMemo(
-    () => historyStore.getTimeline({publishedId: documentId, enableTrace: isDev}),
-    [documentId, historyStore]
-  )
   const [timelineMode, setTimelineMode] = useState<'since' | 'rev' | 'closed'>('closed')
-  // NOTE: this emits sync so can never be null
-  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-  const {historyController} = useMemoObservable(
-    () => historyStore.getTimelineController({client, documentId, documentType, timeline}),
-    [client, documentId, documentType, timeline]
-  )!
```

Comment:
The primary culprit of unnecessary document re-renders, this was previously triggering re-renders on _every_ `handler` callback within `TimelineController`

Reply:
This was also doubling up fetches in strict mode (as `getTimelineController` was also responsible for fetching document pairs, in addition to creating an observable controller)